yb-voyager/src/cp/control_plane.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright (c) YugabyteDB, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package cp
+
+import "github.com/google/uuid"
+
+type ControlPlane interface {
+	Init() error
+	ExportSchemaStarted(*ExportSchemaEvent)
+	ExportSchemaCompleted(*ExportSchemaEvent) // Only success is reported.
+
+	// SubmitMigrationAssessmentReport(*MigrationAssessmentReport)
+
+	SchemaAnalysisStarted(*SchemaAnalysisEvent)
+	SubmitSchemaAnalysisReport(*SchemaAnalysisReport)
+
+	SnapshotExportStarted(*SnapshotExportEvent)
+	UpdateExportedRowCount([]*SnapshotExportTableMetrics)
+	SnapshotExportCompleted(*SnapshotExportEvent)
+
+	// ChangeStreamExportStarted()
+	// UpdateExportedChangesCount(*ExportedChangesCounter)
+
+	ImportSchemaStarted(*ImportSchemaEvent)
+	ImportSchemaCompleted(*ImportSchemaEvent)
+
+	SnapshotImportStarted(*SnapshotImportEvent)
+	UpdateImportedRowCount([]*SnapshotImportTableMetrics)
+	SnapshotImportCompleted(*SnapshotImportEvent)
+
+	Finalize()
+
+	// ChangeStreamImportStarted()
+	// UpdateImportedChangesCount(*ImportedChangesCounter)
+
+	// CutoverInitiated()
+	// CutoverCompleted()
+}
+
+type ExportSchemaEvent struct {
+	MigrationUUID uuid.UUID
+	DatabaseName  string
+	SchemaName    string
+	DBType        string
+}
+
+type SchemaAnalysisEvent struct {
+	MigrationUUID uuid.UUID
+	DatabaseName  string
+	SchemaName    string
+	DBType        string
+}
+
+type SchemaAnalysisReport struct {
+	MigrationUUID uuid.UUID
+	DatabaseName  string
+	SchemaName    string
```

Comment:
Refer https://github.com/yugabyte/yb-voyager/blob/c0896ab4d8ece48c4fff190fd45f30b608dd3f08/yb-voyager/cmd/export.go#L254

In case of postgres, `Schema` is a list of schema names separated by the pipe char. It was a bad decision (taken in very early days of voyager) to represent the list using pipe-separated-string format! You don't have to carry that baggage while designing the `ControlPlane` interface.
You should:
- define schema list as a `SchemaNames []string`; and
- define a `SetSchemaNameList(source.Schema)` method that splits the pipe-separated schema names and assigns to `SchemaNames` as a slice of strings.

Reply:
Changed schemaName to a list.